zyte_common_items/pages.py

```diff
@@ -54,6 +61,36 @@ def _get_metadata_class(cls: type) -> Optional[Type[MetadataT]]:
     return get_generic_param(cls, HasMetadata)
 
 
+class PriceMixin(FieldsMixin):
+    """Provides price-related field implementations."""
+
+    _parsed_price: Optional[Price] = None
+
+    async def _get_parsed_price(self) -> Optional[Price]:
+        if self._parsed_price is None:
+            price_field = getattr(self, "price", object())
+            if price_field is None:
+                # the price field was executed and returned None
+                self._parsed_price = Price(None, None, None)
+                return self._parsed_price
+            # the field wasn't executed yet
+            await ensure_awaitable(price_field)
+        return self._parsed_price
+
+    @field
+    def currency(self) -> Optional[str]:
+        if hasattr(self, "CURRENCY"):
+            return self.CURRENCY
+        return None
```

Comment:
:lipstick: 

```suggestion
        return getattr(self, "CURRENCY", None)
```

I actually verified `tox -e py311` still passes, because I’ve seen weird things about getattr and hasattr happen when attrs is involved :sweat_smile: